### PR TITLE
[INLONG-5598][Manager][DataProxy] Mask sensitive data in logs

### LINKED ISSUE #5598: [Feature][Manager][Dataproxy] Mask sensitive data in logs
### Description

Mask sensitive data in logs

### Use case

Mask sensitive data in logs

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by e-mhui**
### Prepare a Pull Request

- [INLONG-5598][Manager][Dataproxy] Mask sensitive data in logs

- Fixes #5598

### Motivation

Mask sensitive data in logs

### Modifications

Add MaskDataUtils and define a LogMaskConverter to format logged message

### Verifying this change

- Add MaskDataUtils
- Add LogMaskConverter
- Modify log4j.xml

### CODE REVIEW

**Comment by dockerzhang** [comment]
@e-mhui please add UTs for this PR, and add more details about the features.

File: inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    private static final List<String> KEYWORDS =
+            Arrays.asList("password", "passphrase", "pwd", "pass", "secret_id", "token",
+                    "secret_key", "secret_token", "secretKey", "publicKey", "secretKey",
+                    "secretId");
+    private static final List<String> SEPARATORS = Arrays.asList(":", "=", "\": \"", "\":\"");
+    private static final List<Character> STOP_CHARACTERS = Arrays.asList('\'', '"');
+    private static final List<Character> KNOWN_DELIMITERS =
+            Collections.unmodifiableList(Arrays.asList('\'', '"', '<', '>'));
+
+    public static void mask(StringBuilder stringBuilder) {
```
Comment (healchow) [inline inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java:39]: Please add Java docs for static methods.

File: inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+/**
+ * MaskDataUtils is used to mask sensitive information.
+ */
+public class MaskDataUtils {
+
+    private static final List<String> KEYWORDS =
+            Arrays.asList("password", "passphrase", "pwd", "pass", "secret_id", "token",
+                    "secret_key", "secret_token", "secretKey", "publicKey", "secretKey",
+                    "secretId");
```
Comment (healchow) [inline inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java:33]: The `secretKey` was duplicated. Suggested changing to:
```
    private static final List<String> KEYWORDS = Arrays.asList(
            "password", "passphrase", "pwd", "pass",
            "token", "secret_token", "secret_id", "secretId",
            "secret_key", "secretKey", "public_key", "publicKey");
```

File: inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java (hunk 1 above)
Comment (e-mhui) [reply inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java:44]: done

File: inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java (hunk 2 above)
Comment (e-mhui) [reply inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java]: Hello, secretKey and secret_key are different keyword

**Comment by e-mhui** [comment]
> @e-mhui please add UTs for this PR, and add more details about the features.

done

File: inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+/**
+ * MaskDataUtils is used to mask sensitive message in the raw data.
+ */
+public class MaskDataUtils {
+
+    private static final List<String> KEYWORDS = Arrays.asList(
+            "password", "pwd", "pass", "token",
+            "secret_token", "secret_id", "secretId",
+            "secret_key", "secretKey", "publicKey");
```
Comment (healchow) [inline inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java:33]: Does the necessary for the `public_key`?

Comment (e-mhui) [reply inlong-common/src/main/java/org/apache/inlong/common/util/MaskDataUtils.java]: I have remove it

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/config/ControllerExceptionHandler.java [hunk 4]
```diff
@@ -41,18 +42,19 @@
 /**
  * Handler of controller exception.
  */
-@Slf4j
 @RestControllerAdvice
 public class ControllerExceptionHandler {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(ControllerExceptionHandler.class);
```
Comment (leosanqing) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/config/ControllerExceptionHandler.java:48]: Why not use @Slf4j?

File: inlong-common/src/test/java/org/apache/inlong/common/util/MaskDataUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+import org.junit.Test;
+
+/**
+ * MaskDataUtils unit test
+ */
+public class MaskDataUtilsTest {
+
+    @Test
+    public void testMaskDataUtils() throws Exception {
```
Comment (leosanqing) [inline inlong-common/src/test/java/org/apache/inlong/common/util/MaskDataUtilsTest.java:30]: Junit5 does not need to declare public

Comment (e-mhui) [reply inlong-common/src/test/java/org/apache/inlong/common/util/MaskDataUtilsTest.java:30]: what's mean？

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/config/ControllerExceptionHandler.java (hunk 4 above)
Comment (e-mhui) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/config/ControllerExceptionHandler.java]: I saw other module use `LOGGER`, but i don't notice **manager-web** use `@slf4j`. i have correct it.

Comment (leosanqing) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/config/ControllerExceptionHandler.java]: The new code is more @Slf4j

File: inlong-common/src/test/java/org/apache/inlong/common/util/MaskDataUtilsTest.java (hunk 5 above)
Comment (e-mhui) [reply inlong-common/src/test/java/org/apache/inlong/common/util/MaskDataUtilsTest.java:30]: > Junit5 does not need to declare public

it’s not necessary

File: inlong-manager/manager-web/src/test/java/org/apache/inlong/manager/web/config/LogMaskerConverterTest.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test class for LogMaskerConverter.
+ */
+public class LogMaskerConverterTest {
+
+    @Test
+    public void testLogMaskerConverter() {
```
[the remark this answers is not in the document]
Comment (e-mhui) [reply inlong-manager/manager-web/src/test/java/org/apache/inlong/manager/web/config/LogMaskerConverterTest.java:33]: it’s not necessary

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/config/ControllerExceptionHandler.java (hunk 4 above)
Comment (e-mhui) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/config/ControllerExceptionHandler.java]: right, i have correct it.

**Review by healchow — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Review by leosanqing — APPROVED** [review]

**Review by woofyzhao — APPROVED** [review]

**Pull request merged**